src/main/java/sparksoniq/jsoniq/runtime/iterator/postfix/ObjectLookupIterator.java

```diff
@@ -43,58 +43,55 @@ public ObjectLookupIterator(RuntimeIterator object, StringRuntimeIterator string
 
     @Override
     public void open(DynamicContext context) {
-        if(this.isOpen())
-            throw new IteratorFlowException("Variable reference iterator already open", getMetadata());
-        this._currentDynamicContext = context;
-        this._currentIndex = 0;
-        this.results = null;
-        this._hasNext = true;
+        super.open(context);
+        this._nextIndex = 0;
+
+        RuntimeIterator iterator = this._children.get(0);
+        this.items = getItemsFromIteratorWithCurrentContext(iterator);
+
+        this._children.get(1).open(_currentDynamicContext);
+        this._lookupKey = this._children.get(1).next();
```

Comment:
You need to first check this._children.get(1).hasNext(). hasNext() must always be called before next() is called.

Adding a test for {"foo":"bar"}.() should make sure that a clean error is thrown (as exactly one string item is expected on the right).